daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/dfa/CreateDelimiterDFA.scala

```diff
@@ -116,6 +118,7 @@ object CreateDelimiterDFA {
 
   private def buildTransitions(delim: Seq[DelimBase],
     allStates: ArrayBuffer[State], ignoreCase: Boolean): State = {
+    System.out.println("Delim: " + delim)
     assert(!delim.isEmpty)
```

Comment:
This is the Assert call that fails, even after the SDE is called.

Reply:
Changing from the scala assert() method to daffodils Assert.invariant() results in a stack trace that makes it a little easier to see what's going on. This is part of the stack trace:
```
[error] org.apache.daffodil.exceptions.Assert$.abort(Assert.scala:129)
[error] org.apache.daffodil.processors.dfa.CreateDelimiterDFA$.buildTransitions(CreateDelimiterDFA.scala:123)
[error] org.apache.daffodil.processors.dfa.CreateDelimiterDFA$.apply(CreateDelimiterDFA.scala:44)
[error] org.apache.daffodil.processors.dfa.CreateDelimiterDFA$.apply(CreateDelimiterDFA.scala:83), took 1.291 sec
[error]     at org.apache.daffodil.exceptions.Assert$.abort(Assert.scala:129)
[error]     at org.apache.daffodil.processors.dfa.CreateDelimiterDFA$.buildTransitions(CreateDelimiterDFA.scala:123)
[error]     at org.apache.daffodil.processors.dfa.CreateDelimiterDFA$.apply(CreateDelimiterDFA.scala:44)
[error]     at org.apache.daffodil.processors.dfa.CreateDelimiterDFA$.apply(CreateDelimiterDFA.scala:83)
[error]     at org.apache.daffodil.processors.DelimiterUnparseEv.compute(EvDelimiters.scala:100)
[error]     at org.apache.daffodil.processors.DelimiterUnparseEv.compute(EvDelimiters.scala:83)
```
So the assert is getting called from the DelimiterUnparseEv. I suspect you just need to add logic to that Ev, similar to what you do for DelimiterParseEv.

Reply:
Ah, that makes sense.  Will fix.